internal/hrq/hrqreconciler_test.go

```diff
@@ -60,91 +71,103 @@ var _ = Describe("HRQ reconciler tests", func() {
 		Eventually(getHRQUsed(ctx, bazName, bazHRQName)).Should(equalRL("pods", "0"))
 
 		// Increase secret counts from 10 to 11 in baz and verify that the usage is
 		// increased in foo's HRQ. bar's (not an ancestor of baz) and baz' (secrets
 		// is not limited in baz' HRQ) remain unchanged.
-		updateRQUsage(ctx, bazName, "secrets", "11")
+		updateRQUsage(ctx, bazName, api.ResourceQuotaSingletonName, "secrets", "11")
 		Eventually(getHRQUsed(ctx, fooName, fooHRQName)).Should(equalRL("secrets", "11", "pods", "0"))
 		Eventually(getHRQUsed(ctx, barName, barHRQName)).Should(equalRL("secrets", "0", "cpu", "0"))
 		Eventually(getHRQUsed(ctx, bazName, bazHRQName)).Should(equalRL("pods", "0"))
 
 		// Decrease secret counts from 10 to 0 in baz and ensure all usages are gone.
-		updateRQUsage(ctx, bazName, "secrets", "0")
+		updateRQUsage(ctx, bazName, api.ResourceQuotaSingletonName, "secrets", "0")
 		Eventually(getHRQUsed(ctx, fooName, fooHRQName)).Should(equalRL("secrets", "0", "pods", "0"))
 		Eventually(getHRQUsed(ctx, barName, barHRQName)).Should(equalRL("secrets", "0", "cpu", "0"))
 		Eventually(getHRQUsed(ctx, bazName, bazHRQName)).Should(equalRL("pods", "0"))
 	})
 
 	It("should update limits in ResourceQuota objects correctly after deleting an HRQ object", func() {
-		setHRQ(ctx, fooHRQName, fooName, "secrets", "6", "pods", "3")
-		setHRQ(ctx, barHRQName, barName, "secrets", "100", "cpu", "50")
+		setHRQ(ctx, fooHRQName, fooName, nil, "secrets", "6", "pods", "3")
+		setHRQ(ctx, barHRQName, barName, nil, "secrets", "100", "cpu", "50")
 
 		// The RQ in foo should equal its HRQ, while in bar it should be the intersection
-		Eventually(getRQSpec(ctx, fooName)).Should(equalRL("secrets", "6", "pods", "3"))
-		Eventually(getRQSpec(ctx, barName)).Should(equalRL("secrets", "6", "pods", "3", "cpu", "50"))
+		Eventually(getRQHard(ctx, fooName, api.ResourceQuotaSingletonName)).Should(equalRL("secrets", "6", "pods", "3"))
+		Eventually(getRQHard(ctx, barName, api.ResourceQuotaSingletonName)).Should(equalRL("secrets", "6", "pods", "3", "cpu", "50"))
 
 		// After deleting foo's HRQ, bar's RQ should be the same as its HRQ on its own
 		deleteHierarchicalResourceQuota(ctx, fooName, fooHRQName)
-		Eventually(getRQSpec(ctx, fooName)).Should(equalRL())
-		Eventually(getRQSpec(ctx, barName)).Should(equalRL("secrets", "100", "cpu", "50"))
+		Eventually(getRQHard(ctx, fooName, api.ResourceQuotaSingletonName)).Should(equalRL())
+		Eventually(getRQHard(ctx, barName, api.ResourceQuotaSingletonName)).Should(equalRL("secrets", "100", "cpu", "50"))
 	})
 
 	It("should recover if the subtree usages are out of sync in the forest and in reality", func() {
-		setHRQ(ctx, fooHRQName, fooName, "secrets", "100")
-		setHRQ(ctx, barHRQName, barName, "secrets", "100")
+		setHRQ(ctx, fooHRQName, fooName, nil, "secrets", "100")
+		setHRQ(ctx, barHRQName, barName, nil, "secrets", "100")
+		setHRQ(ctx, "hrq-selector", fooName, &highPrioritySelector, "cpu", "4", "pods", "2")
```

Comment:
Hint for reviewers: 
I recommend that you look at this integration test first to understand how it works.